### PR TITLE
Documenting the project (Readme) and it's behaviour (Diagrams)

### PR DESCRIPTION
**Pull request by telliere**
- [ ] Readme 
   - [x]  Quickstart
     - [x] Client
     - [ ] Server
   - [x] Limitations

 - [x] CLI

 - [x] Architecture
   - [x] Global Architecture
   - [x] Sequence Diagrams
     - [x] Data preparation
     - [x] Container preparation
     - [x] Job preparation
   - [ ] Token lifecycle - Permissions

### CODE REVIEW

**Comment by mmatthiesencsc** [comment]
Hi Emilien, thanks a lot for this very detailed documentation, really good work! I suggest we have a quick look through it face2face, but please feel free to accept the suggestions that are easy to accept and don't need discussion.

**Comment by telliere** [comment]
I don't have any suggestion on the PR

**Review by mmatthiesencsc — CHANGES_REQUESTED** [review]

File: docs/architecture/container_preparation.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    HPCS Server ->> Spire Server: Validate SVID
+    Spire Server ->> HPCS Spire Agent: 
+    HPCS Spire Agent ->> Spire Server: Validate SVID
+    Spire Server ->> HPCS Server: 
+    HPCS Server ->> Spire Server: Create workloads identities to access the key
+    Spire Server ->> HPCS Server: 
+    HPCS Server ->> Vault: Create role and policy to access the key
+    Vault ->> HPCS Server: 
+    HPCS Server ->> Container Preparation container: SpiffeID & role to access the container, path to the secret
+    Container Preparation container ->> Container Preparation container: Parse info file based on previous steps
```
[the remark this answers is not in the document]
Comment (telliere) [reply docs/architecture/container_preparation.md:30]: Encryption / Preparation isn't part of the process here, it's encapsulated in "Prepare container, results in an encrypted image and a private key (see below)" -> Reference to sequence diagrams coming after, explaining the prepare & encrypt process

**Comment by telliere** [comment]
Thank you very much for the review @mmatthiesencsc. Really valuable to me. Applied every recommandations after reviewing them

**Comment by telliere** [comment]
Last bits of documentation are here, a last review should allow us to merge this branch @mmatthiesencsc

**Review by mmatthiesencsc — APPROVED** [review]
Thanks Emilien, merging as last state of your efforts this time. I have not yet tested the installation.

**Pull request merged**